### ISSUE TITLE
`declare class` is represented the same as a regular class

### ISSUE THREAD

**Issue by j-f1** [eslint/typescript-eslint-parser#384]
**What version of TypeScript are you using?** 2.4.2

**What version of `typescript-eslint-parser` are you using?** 6.0.1

**What code were you trying to parse?**

```ts
declare class Foo {}
declare var Foo
declare function Foo() {}
```

**What did you expect to happen?**

The AST should specify that the `Foo` class and the `Foo` variable aren’t actual declarations.

Maybe

```ts
// declare var Foo
({
  type: 'VariableDeclaration',
  declarations: 
   [ { type: 'VariableDeclarator',
       id: 
        { type: 'Identifier',
         name: 'Foo' },
      init: null } ],
  kind: 'declare var'
})

// declare class Foo {}
({
  type: 'TSAmbientClass',
  id: 
   { type: 'Identifier',
     name: 'Foo' },
  body: 
   { type: 'ClassBody',
     body: [],
  superClass: null
})
```

**What happened?**

The AST is the same as the one generated for this code:

```ts
class Foo {}
var Foo
declare function Foo() {}
```

Generated AST:

```ts
({ type: 'Program',
  body: 
   [ { type: 'ClassDeclaration',
       id: 
        { type: 'Identifier',
          name: 'Foo' },
       body: 
        { type: 'ClassBody',
          body: [] },
       superClass: null },
     { type: 'VariableDeclaration',
       declarations: 
        [ { type: 'VariableDeclarator',
            id: 
             { type: 'Identifier',
               name: 'Foo' },
            init: null } ],
       kind: 'var' },
     { type: 'DeclareFunction',
       id: 
        { type: 'Identifier',
          name: 'Foo' },
       generator: false,
       expression: false,
       async: false,
       params: [],
       body: 
        { type: 'BlockStatement',
          body: [] } } ],
  sourceType: 'script' })
```

**Comment by j-f1** [comment]
This breaks `no-useless-constructor`.

**Comment by mysticatea** [comment]
I removed `ESLint rule/integration issue` label because this is an AST issue that the AST cannot distinguish semantics. So let's solve it on the underlying package.

/cc @JamesHenry

**Comment by JamesHenry** [comment]
Fxed by #596
